traversal/planner/GraphPlanner.java

```diff
@@ -360,15 +341,15 @@ private void optimise(GraphManager graph, boolean singleUse) {
             return;
         }
 
+        if (!isPlanned()) setInitialValues();
+
         // TODO: we should have a more clever logic to allocate extra time
         long allocatedDuration = singleUse ? HIGHER_TIME_LIMIT_MILLIS : DEFAULT_TIME_LIMIT_MILLIS;
         Instant start, endSolver, end;
         totalDuration += allocatedDuration;
-        solver.setTimeLimit(totalDuration);
 
         start = Instant.now();
-        resultStatus = solver.solve(parameters);
-        resetInitialValues();
```

Comment:
removing the hints from the MPSolver is now implicit after we first call `optimise()`